monai/transforms/spatial/array.py

```diff
@@ -535,6 +535,9 @@ def __call__(
             dtype=dtype,
             lazy=lazy_,
         )
+        if isinstance(data_array, MetaTensor) and "pixdim" in data_array.meta:
+            data_array = cast(MetaTensor, data_array.clone())
```

Comment:
Why is it necessary to clone the data array here? This is going to have a cost and I think isn't compatible with lazy resampling. Perhaps this is code that should be `SpactialResample` instead? @atbenmurray If you could please check if this is going to interact with laziness, thanks.